Create understanding-pe-signatures.md

ericlaw1979 opened the pull request:
To help developers understand that a PE's signature does not cover all bytes of the file, only the code sections.

ericlaw1979 committed:
Create understanding-pe-signatures.md

To help developers understand that a PE's signature does not cover all bytes of the file, only the code sections.

alvinashcraft commented:
Thanks for this, Eric. I'm going to make some minor tweaks just to the markdown formatting. The content looks great.

This should also be added to the toc.yml file in the SecBP folder so it appears in the table of contents on the left, and do you think it makes sense to add a row to link to this topic from the table at the bottom of the [best-practices-for-the-security-apis.md](https://learn.microsoft.com/windows/win32/secbp/best-practices-for-the-security-apis) article at the root of this section?

ericlaw1979 commented:
@alvinashcraft Thanks for your help! After the topic was live, I was going to add links to it from the WinVerifyTrust/WinVerifyTrustEx documentation as well as the page that has the API sample for signature verification. (This particular information is of deep interest to security vendors and not particularly important for general developers).

ericlaw1979 committed:
Update toc.yml

alvinashcraft committed:
Update understanding-pe-signatures.md

Update formatting and add Related content section for Learn Linter pattern compliance.

alvinashcraft commented:
Thanks for this contribution, @ericlaw1979. We have a process that runs nightly to merge changes to the live branch for publishing. So, this new content will be live late tonight PDT.
